lib/galaxy/tools/parameters/basic.py

```diff
@@ -327,6 +327,15 @@ def __init__(self, tool, input_source):
         input_source = ensure_input_source(input_source)
         super().__init__(tool, input_source)
         self.optional = input_source.get_bool("optional", False)
+
+        if not self.optional:
```

Comment:
```suggestion
        if not self.optional and self.type == "text":
```
since other parameter types in the class hierarchy inherit from `TextToolParameter`